#03 AS: added documentation and functionality to generate embeddings from them

AdoshSingh opened the pull request:
- Added mdx file which is consumed as documentation for AI to refer from (these are created from devzero docs).
- Added scripts which are used to generate and store embeddings from the docs.
- Added supabase migrations for initializing the table and creating a function to compare embeddings.

*Note*:  Around 500 lines are auto-generated due to package-lock json on installing some new dependencies.

renatonouman reviewed (approved):
Approved, but can we please add a sentence somewhere that says this documentation is for testing purposes only and is not the real documentation for DevZero?

AdoshSingh commented:
> Approved, but can we please add a sentence somewhere that says this documentation is for testing purposes only and is not the real documentation for DevZero?

I've added a disclaimer sections in the UI explicitly mentioning that the documentation is for testing purposes only, also added disclaimer in the README.md.
